### ISSUE TITLE
Worker Protocol uses invalid JSON

### ISSUE THREAD

**Issue by lukehoersten** [discoproject/disco#275]
As far as I can tell from the [JSON spec](http://www.ietf.org/rfc/rfc4627.txt?number=4627), valid JSON must be either an Object or Array at the root node of the parse tree. Worker protocol messages with simply a number or string are invalid. Again, this is just my unprofessional interpretation of the JSON spec and could be wrong.

> JSON Grammar
> 
> A JSON text is a sequence of tokens.  The set of tokens includes six
> structural characters, strings, numbers, and three literal names.
> 
> A JSON text is a serialized object or array.
> 
> JSON-text = object / array

**Comment by tuulos** [comment]
good point! You are very much right, even based on the original spec at http://json.org

We should change the protocol accordingly in the next revision.

**Comment by lukehoersten** [comment]
All "empty" payload messages are not actually empty. The expected value is actually two quotes (`""`) with payload length 2. Invalid JSON. This includes the `TASK`, `INPUT`, `FAIL`, `DONE`, `PING`, and `OK` messages. These should probably be changed to an empty list (`[]`) or literally just a truly empty payload -- no JSON at all.
